great_expectations/core/util.py

```diff
@@ -782,148 +771,150 @@ def sniff_s3_compression(s3_url: S3Url) -> Union[str, None]:
+    # deprecated-v1.0.0
+    warnings.warn(
+        "Utility method get_or_create_spark_application() is deprecated and will be removed in v1.0.0. "
+        "Please use the method great_expectations.core.get_or_create_spark_session() instead.",
+        category=DeprecationWarning,
+    )
+    if force_reuse_spark_context is not None:
+        # deprecated-v1.0.0
+        warnings.warn(
+            "force_reuse_spark_context is deprecated and will be removed in version 1.0. "
+            "In environments that allow it, the existing Spark context will be reused, adding the "
+            "spark_config options that have been passed. If the Spark context cannot be updated with "
+            "the spark_config, the context will be stopped and restarted with the new spark_config.",
+            category=DeprecationWarning,
+        )
+    return get_or_create_spark_session(spark_config=spark_config)
+
 
-    Due to the uniqueness of SparkContext per JVM, it is impossible to change SparkSession configuration dynamically.
-    Attempts to circumvent this constraint cause "ValueError: Cannot run multiple SparkContexts at once" to be thrown.
-    Hence, SparkSession with SparkConf acceptable for all tests must be established at "pytest" collection time.
-    This is preferred to calling "return SparkSession.builder.getOrCreate()", which will result in the setting
-    ("spark.app.name", "pyspark-shell") remaining in SparkConf statically for the entire duration of the "pytest" run.
+def get_or_create_spark_session(
+    spark_config: Optional[dict[str, str]] = None,
+) -> pyspark.SparkSession:
+    """Obtains Spark session if it already exists; otherwise creates Spark session and returns it to caller.
 
     Args:
         spark_config: Dictionary containing Spark configuration (string-valued keys mapped to string-valued properties).
-        force_reuse_spark_context: Boolean flag indicating (if True) that creating new Spark context is forbidden.
 
-    Returns: SparkSession (new or existing as per "isStopped()" status).
+    Returns:
+        SparkSession
     """
-    if spark_config is None:
-        spark_config = {}
-    else:
-        spark_config = copy.deepcopy(spark_config)
+    spark_config = spark_config or {}
 
-    name: Optional[str] = spark_config.get("spark.app.name")
-    if not name:
-        name = "default_great_expectations_spark_application"
-
-    spark_config.update({"spark.app.name": name})
-
-    spark_session: Optional[pyspark.SparkSession] = get_or_create_spark_session(
-        spark_config=spark_config
+    spark_session_type: type[pyspark.SparkSession | pyspark.SparkConnectSession]
+    spark_session: pyspark.SparkSession
+    try:
+        spark_session_type = pyspark.SparkSession
+        spark_session = spark_session_type.builder.getOrCreate()
+    except RuntimeError:
+        spark_session_type = pyspark.SparkConnectSession
+        spark_session = spark_session_type.builder.getOrCreate()
```

Comment:
Can this raise an exception?

Reply:
It could. If it does I think we should raise the initial `RuntimeError` triggered on line 819. ✅